src/internal.c

```diff
@@ -9377,16 +9377,43 @@ int CheckAltNames(DecodedCert* dCert, char* domain)
 {
     int        match = 0;
     DNS_entry* altName = NULL;
+    char *buf;
+    int len;
 
     WOLFSSL_MSG("Checking AltNames");
 
     if (dCert)
         altName = dCert->altNames;
 
     while (altName) {
+#if defined(WOLFSSL_IP_ALT_NAME)
+        char name[WOLFSSL_MAX_IPSTR] = {0};
+        /* check if alt name is stored as IP addr octet */
+        if (altName->type == ASN_IP_TYPE) {
+            char tmp[4];
+            int i;
+            word32 idx = 0;
+            for (i = 0; i < altName->len; i++) {
+                XMEMSET(tmp, 0, sizeof(tmp));
+                XSNPRINTF(tmp, sizeof(tmp), (altName->len <= 4) ? "%u" : "%02X",
+                        altName->name[i]);
+                XSTRNCAT(&name[idx], tmp, sizeof(tmp)-1);
+                idx += XSTRLEN(tmp);
```

Comment:
There should be a check on idx size compared to WOLFSSL_MAX_IPSTR to avoid reading/writing over the edge of the buffer